src/reader.cpp

```diff
@@ -1501,8 +1502,12 @@ bool reader_data_t::expand_abbreviation_at_cursor(size_t cursor_backtrack) {
         size_t cursor_pos = el->position() - std::min(el->position(), cursor_backtrack);
         if (auto replacement =
                 reader_expand_abbreviation_at_cursor(el->text(), cursor_pos, this->parser())) {
-            push_edit(el, edit_t{replacement->range, std::move(replacement->text)});
-            update_buff_pos(el, replacement->cursor);
+            push_edit(el, edit_t{replacement->range, *replacement->text});
+            if (replacement->has_cursor) {
```

Comment:
This commit fixes a bug introduced by the first commit.
It'd be better to squash it, so the bug is never introduced.
As a reviewer I don't want to know how the sausage is made.
Also, every commit should pass tests.  (This means he formatting fixes should be placed first etc.)
See also https://www.bitsnbites.eu/git-history-work-log-vs-recipe/